lib/WUI/link_content/prusa_link_api.cpp

```diff
@@ -89,18 +89,52 @@ optional<ConnectionState> PrusaLinkApi::accept(const RequestParser &parser) cons
              * transformation into a file name (removal of query params,
              * forbidding of '..', etc).
              */
+            /*
+             * TODO:
+             * We currently don't read the URI parameters. That is, we don't
+             * understand "?recursive=true" and therefore don't do recursive.
+             *
+             * The FileInfo handler doesn't know recursive yet either.
+             */
             static const auto prefix = "/api/files";
             static const size_t prefix_len = strlen(prefix);
             char fname[FILE_PATH_BUFFER_LEN + prefix_len];
             if (parser.uri_filename(fname, sizeof fname)) {
                 size_t len = strlen(fname);
-                const char *fname_real;
                 if (prefix_len > len) {
                     return StatusPage(Status::NotFound, parser.can_keep_alive());
-                } else if (prefix_len == len) {
-                    fname_real = "/";
-                } else {
-                    fname_real = fname + prefix_len;
+                }
+
+                const char *fname_real = fname + prefix_len;
+
+                /*
+                 * The octoprint API gives special meaning to /local and
+                 * /sdcard. For us, everything lives in the USB (/usb). We
+                 * remap these. Nevertheless, we never _generate_ these /local
+                 * or such URIs, so we don't remap anything else but the "root".
+                 */
+                static const char *const roots[] = {
+                    "",
+                    "/",
+                    "/local",
+                    "/local/",
+                    "/sdcard",
+                    "/sdcard/",
+                };
+
+                for (size_t i = 0; i < sizeof roots / sizeof *roots; i++) {
```

Comment:
nitpick: I prefer seeing:
`sizeof(roots) / sizeof(roots[0])`
but ideally add a template for such a construct like this:
```
template <typename T> const size_t arrsz(const T &aObj){     
    return sizeof(aObj)/sizeof aObj[0];
}
```

Reply:
I'll change it (though the `*` variant seems more nuanced to `arbitrary element` over `first element`; but let's not go into that discussion 😈).

I'd be a bit vary about the template; I don't remember the exact rules when an array decays to a pointer when passed as a parameter, in which case it could specialize as `arrsz(const Whatever &*aObj)`, which would always return 1… maybe it doesn't, in this case, but...

Reply:
ok